packages/react-native/ReactAndroid/src/main/java/com/facebook/react/bridge/JavaMethodWrapper.kt

```diff
@@ -0,0 +1,403 @@
+  private val paramLength: Int
+
+  /**
+   * Determines how the method is exported in JavaScript: METHOD_TYPE_ASYNC for regular methods
+   * METHOD_TYPE_PROMISE for methods that return a promise object to the caller. METHOD_TYPE_SYNC
+   * for sync methods
+   */
+  override var type: String = BaseJavaModule.METHOD_TYPE_ASYNC
+  private var argumentsProcessed = false
+  private var argumentExtractors: Array<ArgumentExtractor<*>>? = null
+  private var internalSignature: String? = null
+  private var arguments: Array<Any?>? = null
+  private var jsArgumentsNeeded = 0
+
+  init {
+    method.isAccessible = true
+    parameterTypes = method.parameterTypes
+    paramLength = parameterTypes.size
+
+    if (isSync) {
+      type = BaseJavaModule.METHOD_TYPE_SYNC
+    } else if (paramLength > 0 && (parameterTypes[paramLength - 1] == Promise::class.java)) {
+      type = BaseJavaModule.METHOD_TYPE_PROMISE
+    }
+  }
+
+  private fun processArguments() {
+    if (argumentsProcessed) {
+      return
+    }
+    SystraceMessage.beginSection(TRACE_TAG_REACT, "processArguments")
+      .arg("method", moduleWrapper.name + "." + method.name)
+      .flush()
+    try {
+      argumentsProcessed = true
+      argumentExtractors = buildArgumentExtractors(parameterTypes)
+      internalSignature =
+          buildSignature(method, parameterTypes, (type == BaseJavaModule.METHOD_TYPE_SYNC))
+      // Since native methods are invoked from a message queue executed on a single thread, it is
+      // safe to allocate only one arguments object per method that can be reused across calls
+      arguments = arrayOfNulls(parameterTypes.size)
+      jsArgumentsNeeded = calculateJSArgumentsNeeded()
+    } finally {
+      SystraceMessage.endSection(TRACE_TAG_REACT).flush()
+    }
+  }
+
+  val signature: String?
+    get() {
+      if (!argumentsProcessed) {
+        processArguments()
+      }
+      return checkNotNull(internalSignature)
+    }
+
+  private fun buildSignature(
+      method: Method,
+      paramTypes: Array<Class<*>>,
+      isSync: Boolean
+  ): String {
+    val builder = StringBuilder(paramTypes.size + 2)
+
+    if (isSync) {
+      builder.append(returnTypeToChar(method.returnType))
+      builder.append('.')
+    } else {
+      builder.append("v.")
+    }
+
+    for (i in paramTypes.indices) {
+      val paramClass = paramTypes[i]
+      if (paramClass == Promise::class.java) {
+        check(i == paramTypes.size - 1) { "Promise must be used as last parameter only" }
+      }
+      builder.append(paramTypeToChar(paramClass))
+    }
+
+    return builder.toString()
+  }
+
+  private fun buildArgumentExtractors(paramTypes: Array<Class<*>>): Array<ArgumentExtractor<*>> {
+    val argumentExtractors = arrayOfNulls<ArgumentExtractor<*>>(paramTypes.size)
+    var i = 0
+    while (i < paramTypes.size) {
+      val argumentClass = paramTypes[i]
+      val extractor: ArgumentExtractor<*> = when (argumentClass) {
+        java.lang.Boolean::class.java,
```

Comment:
```suggestion
        Boolean::class. javaObjectType,
```
You can use Kotlin type classes and don't need to specify java.lang classes and just need be careful with `javaObjectType` vs `javaPrimitiveType`.
Same for other lines that does type checking.

See this [commit](https://github.com/facebook/react-native/commit/1fb04e2134d042332c2ff5d10b979afde41ab6f6) for more information.

Reply:
Thanks for clarifying this @alanleedev! I did the change now and it works good as well (the original Android Studio conversion had some crashes so the adjustments I did were based on a similar comparison in [TurboModuleInteropUtils.kt](https://github.com/facebook/react-native/blob/main/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/internal/turbomodule/core/TurboModuleInteropUtils.kt#L236))